File: src/main/Java/boj_step9/boj2581.java
import java.io.*;
import java.math.BigInteger;

public class boj2581 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        long m = Long.parseLong(reader.readLine());
        long n = Long.parseLong(reader.readLine());
        long min = BigInteger.valueOf(m).nextProbablePrime().longValue(), sum = 0L;
        if (BigInteger.valueOf(m).isProbablePrime(10)) {
            min = m;
            sum = m;
        }
        while (m <= n) {
            m = BigInteger.valueOf(m).nextProbablePrime().longValue();
            if (m > n) break;
            sum += m;
        }
        if (sum == 0L) writer.append("-1").close();
        else {
            writer.append(String.valueOf(sum));
            writer.newLine();
            writer.append(String.valueOf(min)).close();
        }
    }
}

File: src/main/boj_step9/boj2581.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import kotlin.math.sqrt

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val m = r.readLine().toInt()
    val n = r.readLine().toInt()
    val primes = mutableListOf<Int>()
    for (i in m..n) {
        if (i == 1) continue
        val sqrt = sqrt(i.toDouble()).toInt()
        var isPrime = true
        for (j in 2..sqrt) if (i % j == 0) {
            isPrime = false
            break
        }
        if (isPrime) primes.add(i)
    }
    if (primes.size == 0) w.write("-1")
    else {
        w.write(primes.sum().toString() + "\n")
        w.write(primes.min().toString())
    }
    w.close()
}